lib/Dialect/FIRRTL/Ops.cpp

```diff
@@ -433,8 +433,8 @@ static ParseResult parseFModuleOp(OpAsmParser &parser, OperationState &result,
 
   // Parse the optional function body.
   auto *body = result.addRegion();
-  if (parser.parseOptionalRegion(
-          *body, entryArgs, entryArgs.empty() ? ArrayRef<Type>() : argTypes))
+  if (parser.parseRegion(*body, entryArgs,
```

Comment:
This isn't the right fix.  This "parseRegion" call needs to only happen `if (!isExtModule)`.  Something like:

```
if (!isExtModule) {
  if (parser.parseRegion(*body, entryArgs, ...)) return failure();
}
```

In the extmodule case, we want to "addRegion" but not fill it in.  Same thing on the RTL side of things.

Reply:
ah, the RTL side looks right!